java/code/src/com/redhat/rhn/domain/channel/Channel.hbm.xml

```diff
@@ -145,6 +145,42 @@ PUBLIC "-//Hibernate/Hibernate Mapping DTD 3.0//EN"
         -->
     </class>
 
+    <sql-query name="Channel.findAllByUserOrderByChild">
+        <![CDATA[
+    SELECT {c.*}
+    FROM rhnChannel {c}
+        LEFT OUTER JOIN rhnChannel parent ON c.parent_channel = parent.id
+        LEFT OUTER JOIN rhnChannelCloned c_1_ ON c.id = c_1_.id
+    WHERE EXISTS (
+        SELECT 1
+        FROM suseChannelUserRoleView scur
+        WHERE scur.channel_id = c.id
+        AND scur.user_id = :userId
+        AND scur.deny_reason IS NULL
+    )
+    AND (
+        c.parent_channel IS NULL
+        OR (
+            c.parent_channel IS NOT NULL
+            AND EXISTS (
+                SELECT 1
+                FROM suseChannelUserRoleView scur
+                WHERE scur.channel_id = c.parent_channel
+                AND scur.user_id = :userId
+                AND scur.deny_reason IS NULL
+            )
+        )
+    )
+    ORDER BY
+        c.org_id NULLS FIRST,
+        CASE WHEN c.parent_channel IS NULL THEN c.label
+             WHEN c.parent_channel IS NOT NULL THEN parent.label END,
```

Comment:
maybe we can remote this second case and use `else` instead:
```suggestion
             else parent.label END,
```

Reply:
@rjmateus I was able to simplify it by doing `COALESCE(parent.label, c.label)`, let me know if you see any issue with this approach, thanks.

Reply:
Looks good to me